src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -90,7 +90,19 @@ bool IsSuppressedOnElement (int id, ICustomAttributeProvider provider, out Suppr
 			if (provider == null)
 				return false;
 
-			return _suppressions.TryGetValue (provider, out var suppressions) &&
+			if (_suppressions.TryGetValue (provider, out var suppressions))
+				return suppressions.TryGetValue (id, out info);
+
+			if (!_context.CustomAttributes.HasAny (provider))
+				return false;
+
+			foreach (var ca in _context.CustomAttributes.GetCustomAttributes (provider)) {
+				if (TypeRefHasUnconditionalSuppressions (ca.Constructor.DeclaringType) &&
+					provider is not ModuleDefinition or AssemblyDefinition)
+					AddSuppression (ca, provider);
+			}
+
+			return _suppressions.TryGetValue (provider, out suppressions) &&
 				suppressions.TryGetValue (id, out info);
```

Comment:
Do we have to do this a second time? Can't we keep info of whether we saw any suppression on the above loop and return that?

Reply:
We would need to track the decoded info as we go in addition to whether we saw any so it would look something like this: https://github.com/sbomer/linker/compare/suppressionsFix...sbomer:suppressionsFixFeedback?expand=1. I personally find this less readable but LMK which you prefer.

Reply:
I think it's OK to probe the cache twice. We could only call the `suppression.TryGetValue` once, but it's not a big deal.

Reply:
Thanks for the example! I agree it's not a big deal and makes it less readable. I'm fine with either approach.